src/lurk/cli/microchain.rs

```diff
@@ -274,6 +262,45 @@ impl MicrochainArgs {
     }
 }
 
+fn dump_microchain_data<T: Serialize + ?Sized>(id: &[F], name: &str, data: &T) -> Result<()> {
+    let hash = format!("{:x}", field_elts_to_biguint(id));
+    let dir = microchains_dir()?.join(hash);
+    std::fs::create_dir_all(&dir)?;
+    std::fs::write(dir.join(name), bincode::serialize(data)?)?;
+    Ok(())
+}
+
+fn dump_genesis(id: &[F], genesis: &Genesis) -> Result<()> {
+    dump_microchain_data(id, "genesis", genesis)
+}
+
+fn dump_proofs(id: &[F], proofs: &[OpaqueChainProof]) -> Result<()> {
+    dump_microchain_data(id, "proofs", proofs)
+}
+
+fn dump_state(id: &[F], state: &ChainState) -> Result<()> {
+    dump_microchain_data(id, "state", state)
+}
```

Comment:
nit: Renaming the functions to `save_` instead of `dump_` might be nicer (but it's a matter of opinion)

Reply:
I'm reusing the terminology from other places in the repo (e.g. `dump-expr`)